src/physics/utils/physics_data.F90

```diff
@@ -348,19 +349,24 @@ subroutine check_field_2d(file, var_names, timestep, current_value,        &
       real(kind_phys)                  :: diff
       integer                          :: col
       integer                          :: ierr      !For MPI
+      integer                          :: mpi_stat(mpi_status_size) !For MPI
       real(kind_phys), allocatable     :: buffer(:)
       integer                          :: diff_count
-      real(kind_phys)                  :: max_diff
-      real(kind_phys)                  :: max_diff_gl
+      real(kind_phys)                  :: max_diff(2,1)    !Stores the local max diff and its MPI rank
```

Comment:
why the second dimension (for this and max_diff_gl)? Some MPI necessity?

Reply:
Good question!  It's because I don't know how to read MPI documentation properly.  Extraneous dimension removed.